### ISSUE TITLE
predicate-class doesn't exist on hackage, build issues

### ISSUE THREAD

**Issue by leshow** [uber/queryparser#25]
I read the blog post and wanted to try this out, but I'm running into some issues adding it as a dependency.

A recent PR allowed newer versions of base. But the queryparser on hackage hasn't been updated since then, and it looks like all the builds failed (not sure when this ran- https://hackage.haskell.org/package/queryparser-0.1.0.0/reports/). 

I'm trying to use this package in a stack project and running into issues, `predicate-class` doesn't exist on hackage, and queryparser appears to need it.

I can upload a test project that shows the build failing if that helps.

**Comment by taktoa** [comment]
Perhaps someone should add a `cabal.project` file so that people can more easily build this using `cabal-install` (I've often found Stack to be rather flaky).
[Relevant documentation](http://cabal.readthedocs.io/en/latest/nix-local-build.html#developing-multiple-packages).

**Comment by h4v0kh3l1** [comment]
Hey @leshow and @taktoa,

Thanks for your interest! We haven't pushed the latest queryparser version to hackage yet, though I will find some time this week to do so. Predicate-class, I can push up to hackage as well. That should be enough to get importing from hackage working.

In the future, I'd like to nail down a good push schedule, without flooding hackage's version archive. Still figuring that out.

@taktoa, I'm not familiar with cabal. My understanding, which is probably flawed, was that stack builds atop cabal, so that if you had a stack project, cabal would work as well. What are you missing that would make your setup work?

**Comment by leshow** [comment]
@h4v0kh3l1 That would be awesome, thanks very much.

If you're looking for some help testing once everything is uploaded I'd be happy to lend a hand, let me know what I can do to help

**Comment by h4v0kh3l1** [comment]
Heads up, just pushed a the predicate-class package, along with the other packages to get the latest pull requests in.

Let me know how your testing goes, I'd like to make sure this works for you.

**Comment by leshow** [comment]
I'm not sure why, but in addition to adding `queryparser` and `queryparser-vertica` to my cabal file, I had to add:

```
- queryparser-0.1.0.1
- queryparser-vertica-0.1.0.1
- fixed-list-0.1.6
- predicate-class-0.1.0.1
```

to the stack.yaml. However, it does appear to be building with imported types and functions (though I haven't written anything useful yet)

Thank you very much! :+1:
